mgmtd/mgmt_be_adapter.c

```diff
@@ -34,55 +35,31 @@
  */
 struct mgmt_be_xpath_map {
 	char *xpath_regexp;
-	uint subscr_info[MGMTD_BE_CLIENT_ID_MAX];
-};
-
-struct mgmt_be_client_xpath {
-	const char *xpath;
-	uint subscribed;
-};
-
-struct mgmt_be_client_xpath_map {
-	struct mgmt_be_client_xpath *xpaths;
-	uint nxpaths;
-};
-
-struct mgmt_be_get_adapter_config_params {
-	struct mgmt_be_client_adapter *adapter;
-	struct nb_config_cbs *cfg_chgs;
-	uint32_t seq;
+	uint64_t clients;
 };
 
 /*
  * Each client gets their own map, but also union all the strings into the
  * above map as well.
  */
 #if HAVE_STATICD
-static struct mgmt_be_client_xpath staticd_xpaths[] = {
-	{
-		.xpath = "/frr-vrf:lib/*",
-		.subscribed = MGMT_SUBSCR_VALIDATE_CFG | MGMT_SUBSCR_NOTIFY_CFG,
-	},
-	{
-		.xpath = "/frr-interface:lib/*",
-		.subscribed = MGMT_SUBSCR_VALIDATE_CFG | MGMT_SUBSCR_NOTIFY_CFG,
-	},
-	{
-		.xpath =
-			"/frr-routing:routing/control-plane-protocols/control-plane-protocol/frr-staticd:staticd/*",
-		.subscribed = MGMT_SUBSCR_VALIDATE_CFG | MGMT_SUBSCR_NOTIFY_CFG,
-	},
+static const char *const staticd_xpaths[] = {
```

Comment:
I see that you removed the asterisks from xpaths. I didn't look into the code yet, but want to understand how do you solve the problem of partial subscription only for a subtree? For example, OSPF and ISIS augment interfaces with their specific nodes. I my understanding, they would want to subscribe to the interface list itself and to their specific subtrees, but not to subtrees of another daemon. If asterisks are supported, it would look like this (OSPF for example):
```
"frr-interface:lib/interface" - without asterisk, subscribe to the list itself without children
"frr-interface:lib/interface/osfp/*" - with asterisk, subscribe to the whole OSPF subtree
```

Is this solved somehow in this PR? Or should we subscribe to the whole interface tree in both daemons and just ignore the unknown nodes on the backend?

Reply:
Just leave off the `*`. It's redundant. In your example:

`"frr-interface:lib/interface/ospf"`

Or in a local tree of mine: `"/frr-interface:lib/interface/state/frr-ospfd-lite:ospf/state"` for state.

Reply:
Just to confirm that I understand this correctly. If a daemon subscribes to subtree `"frr-interface:lib/interface/ospf"`, it receives updates for all its parents (`interface` list in this case) as well as subtree itself with all its children. But no to other subtrees or leaf children of any parent, right?

I'm asking, because right now I see that daemons receive updates for the whole tree even if they subscribe only to a subtree. For example, if I subscribe to `"frr-interface:lib/interface/ospf"`, I receive updates for all children of `"frr-interface:lib/interface"`, including `"frr-interface:lib/interface/isis"`, etc. Xpaths here are just examples, they are not real xpaths used in FRR.

Reply:
Yes, this is what I meant by the current code is buggy, and yes it fixes it. Daemons should always filter unnecessary xpaths too, but this is turning off the firehose that existed prior.